bot/exts/utils/thread_bumper.py

```diff
@@ -0,0 +1,145 @@
+import typing as t
+
+import discord
+from async_rediscache import RedisCache
+from discord.ext import commands
+
+from bot import constants
+from bot.bot import Bot
+from bot.log import get_logger
+from bot.pagination import LinePaginator
+from bot.utils import scheduling
+
+log = get_logger(__name__)
+
+
+class ThreadBumper(commands.Cog):
+    """Cog that allow users to add the current thread to a list that get reopened on archive."""
+
+    # RedisCache[discord.Thread.id, "sentinel"]
+    threads_to_bump = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.init_task = scheduling.create_task(self.ensure_bumped_threads_are_active(), event_loop=self.bot.loop)
+
+    async def unarchive_if_not_manually_archived(self, threads: list[discord.Thread]) -> None:
+        """
+        Iterate through and unarchive any threads that weren't manually archived recently.
+
+        This is done by extracting the manually archived threads from the audit log.
+
+        Only the last 200 thread_update logs are checked,
+        as this is assumed to be more than enough to cover bot downtime.
+        """
+        guild = self.bot.get_guild(constants.Guild.id)
+
+        recent_manually_archived_thread_ids = []
+        for thread_update in guild.audit_logs(limit=200, action=discord.AuditLogAction.thread_update):
```

Comment:
[`Guild.audit_logs()`](https://discordpy.readthedocs.io/en/stable/api.html?highlight=audit_logs#discord.Guild.audit_logs) is an async iterator.
```suggestion
        async for thread_update in guild.audit_logs(limit=200, action=discord.AuditLogAction.thread_update):
```

Reply:
hah, you're right, I had this as async locally before I refactored the function and missed this :(

Reply:
https://github.com/python-discord/bot/pull/2084/commits/ef2717706458de4da1441726a0eed1a42bababe8